src/main/java/com/ibm/crypto/plus/provider/AESKeyFactory.java

```diff
@@ -84,7 +84,7 @@ protected KeySpec engineGetKeySpec(SecretKey key, Class<?> keySpec)
                 && (key.getFormat().equalsIgnoreCase("RAW"))) {
 
             // Check if requested key spec is amongst the valid ones
-            if (SecretKeySpec.class.isAssignableFrom(keySpec)) {
+            if (keySpec.isAssignableFrom(SecretKeySpec.class)) {
```

Comment:
In this scenario where there is no cast, do we want to check if `keySpec` is a parent of `SecretKeySpec`, or the other way around? I would think we want `keySpec` to be some subclass of `SecretKeySpec` or even `instanceOf(SecretKeySpec)`, right?

Reply:
Or maybe I'm wrong, cause whoever gets the result of this method will try to cast it into the `keySpec` class probably. It is a bit confusing. @jasonkatonica What do you think?

Reply:
I have the same concern before when dealing with those without cast. 

1. I check the OpenJDK codes which update the `isAssignableFrom` checks (https://github.com/openjdk/jdk/pull/7037/files#diff-743b02b03ca8fa1bda2310721348024c26e1d796c83a7671b03e9332678a10d4R246). And it did the same.

2. From my understanding (Please correct me if I am wrong). It did not explicitly cast, but it `return new SecretKeySpec(key.getEncoded(), key.getAlgorithm()); `which will, in some kind of implicitly way, to cast as the KeySpec.

Reply:
`KeySpec` is an interface that is implemented by `SecretKeySpec`, so there is no need for a cast there. What I, also, noticed though is that `SecretKeySpec` has no parents, other than `Object`. So this check practically verifies whether the `keySpec` parameter is the same class as `SecretKeySpec`.